src/solidity/utils.ts

```diff
@@ -32,26 +32,35 @@ export function solidifyName(fileName: string): string {
 
 export async function gatherSources(
   what: string,
-  workingDir?: string,
+  workingDir: string = process.cwd(),
   resolver: ResolverEngine<ImportFile> = SolidityImportResolver(),
 ): Promise<ImportFile[]> {
   let result: ImportFile[] = [];
-  let queue = [];
+  let queue: { cwd: string; file: string }[] = [];
   let alreadyImported = new Set();
 
-  queue.push({ cwd: workingDir, file: what });
-  alreadyImported.add(solidifyName(what));
+  const absoluteWhat = path.resolve(workingDir, what);
+  queue.push({ cwd: workingDir, file: absoluteWhat });
+  alreadyImported.add(solidifyName(absoluteWhat));
   while (queue.length > 0) {
     const fileData = queue.shift()!;
     const resolvedFile: ImportFile = await resolver.require(fileData.file, fileData.cwd);
     const foundImports = findImports(resolvedFile);
-    result.push(resolvedFile);
-    const filewd = path.dirname(resolvedFile.path);
+
+    // if imported path starts with '.' we assume it's relative and return it's absolute path
+    // if not - return the same name it was imported with
+    if (fileData.file[0] === ".") {
```

Comment:
Not enough if the file is used in Windows context
https://www.npmjs.com/package/is-relative

Reply:
Additionally, @Crypto-Punkers/developers it might be worth debating whether we always want to return an absolute path from the resolver in the first place

Reply:
As per solidity documentation: "All path names are treated as absolute paths unless they start with the current . or the parent directory .. ."
https://solidity.readthedocs.io/en/v0.5.2/layout-of-source-files.html#paths